tools/data_integrity_check.php

```diff
@@ -0,0 +1,129 @@
+<?php declare(strict_types=1);
+/**
+ * This tool verifies that the data contained in the LORIS database is well-formed
+ * and contains no major issues. For example, it checks that tables referenced
+ * in one part of the database exist elsewhere.
+ */
+require_once __DIR__ . '/generic_includes.php';
+require_once __DIR__ . '/cli_helper.class.inc';
+
+
+$helper = new CLI_Helper();
+
+// Verifies that instruments registered in the test_names table have
+// corresponding per-instrument tables containing the data.
+$instrument_arr = $DB->pselectCol("SELECT Test_name FROM test_names", []);
+$check_arr      = [];
+
+$helper->printSuccess('Getting data from `test_names` table...');
+$instruments       = array_filter(
+    $instrument_arr,
+    [$DB, 'tableExists']
+);
+$numberOfTestNames = count($instrument_arr);
+$numberOfTestNamesWithTables = count($instruments);
+if ($numberOfTestNames !== $numberOfTestNamesWithTables) {
+    $helper->printError(
+        'Found entries in the `test_names` table that do not have appear to '
+        . 'be installed: ' . join(', ', array_diff($instrument_arr, $instruments))
+    );
+}
```

Comment:
this check is incorrect now, instruments can be JSONdata and not require a table

Reply:
@johnsaigle this check should still be removed. JSON instruments don't need tables. or it should be improved to check if the instrument has the JSONdata flag to true before reporting an error.

Reply:
How can I determine whether an instrument has JSON data?